packages/forwarder-helper/src/forwarder_helper_factory.ts

```diff
@@ -22,4 +31,220 @@ export const forwarderHelperFactory = {
+        assert.isETHAddressHex('takerAddress', takerAddress);
+        assert.isWebUri('sraUrl', sraUrl);
+        if (!_.isUndefined(rpcUrl)) {
+            assert.isWebUri('rpcUrl', rpcUrl);
+        }
+        assert.isNumber('networkId', networkId);
+        // create provider
+        const providerEngine = new Web3ProviderEngine();
+        if (!_.isUndefined(rpcUrl)) {
+            providerEngine.addProvider(new RPCSubprovider(rpcUrl));
+        }
+        providerEngine.start();
+        // create contract wrappers given provider and networkId
+        const contractWrappers = new ContractWrappers(providerEngine, { networkId });
+        // find ether token asset data
+        const etherTokenAddressIfExists = contractWrappers.etherToken.getContractAddressIfExists();
+        if (_.isUndefined(etherTokenAddressIfExists)) {
+            throw new Error(ForwarderHelperFactoryError.NoEtherTokenContractFound);
+        }
+        const etherTokenAssetData = assetDataUtils.encodeERC20AssetData(etherTokenAddressIfExists);
+        // find zrx token asset data
+        let zrxTokenAssetData: string;
+        try {
+            zrxTokenAssetData = contractWrappers.exchange.getZRXAssetData();
+        } catch (err) {
+            throw new Error(ForwarderHelperFactoryError.NoZrxTokenContractFound);
+        }
+        // get orderbooks for makerAsset/WETH and ZRX/WETH
+        const sraClient = new HttpClient(sraUrl);
+        const orderbookRequests = [
+            { baseAssetData: makerAssetData, quoteAssetData: etherTokenAssetData },
+            { baseAssetData: zrxTokenAssetData, quoteAssetData: etherTokenAssetData },
+        ];
+        const requestOpts = { networkId };
+        // TODO: try catch these requests and throw a more domain specific error
+        const [makerAssetOrderbook, zrxOrderbook] = await Promise.all(
+            _.map(orderbookRequests, request => sraClient.getOrderbookAsync(request, requestOpts)),
+        );
+        // validate orders and find remaining fillable from on chain state or sra api
+        let ordersAndRemainingFillableMakerAssetAmounts: OrdersAndRemainingFillableMakerAssetAmounts;
+        let feeOrdersAndRemainingFillableMakerAssetAmounts: OrdersAndRemainingFillableMakerAssetAmounts;
+        if (!_.isUndefined(rpcUrl)) {
+            // if we do have an rpc url, get on-chain orders and traders info via the OrderValidatorWrapper
+            const ordersFromSra = _.map(makerAssetOrderbook.asks.records, apiOrder => apiOrder.order);
+            const feeOrdersFromSra = _.map(zrxOrderbook.asks.records, apiOrder => apiOrder.order);
+            // TODO: try catch these requests and throw a more domain specific error
+            const [makerAssetOrdersAndTradersInfo, feeOrdersAndTradersInfo] = await Promise.all(
+                _.map([ordersFromSra, feeOrdersFromSra], ordersToBeValidated => {
+                    const takerAddresses = _.map(ordersToBeValidated, () => takerAddress);
```

Comment:
change this to NULL_ADDRESS in conjunction w/ removing takerAddress from the function args